applications/new_schema/types.py

```diff
@@ -64,6 +65,12 @@ class BerthApplicationFilter(django_filters.FilterSet):
         fields=(("created_at", "createdAt"),),
         label="Supports only `createdAt` and `-createdAt`.",
     )
+    name = django_filters.CharFilter(method="filter_name")
+
+    def filter_name(self, qs, name, value):
+        for part in value.split():
+            qs.filter(Q(first_name__icontains=part) | Q(last_name__icontains=part))
```

Comment:
You need to assign the filtered queryset so that it takes effect

```suggestion
            qs = qs.filter(Q(first_name__icontains=part) | Q(last_name__icontains=part))
```

Reply:
👍 Noticed this as well, fixed and added test for case where nothing should match.